src/mtcp/mtcp_restart.c

```diff
@@ -738,13 +781,18 @@ main(int argc, char *argv[], char **environ)
 #endif
     typedef int (*getRankFptr_t)(void);
     int rank = -1;
+    /* FIXME: combine GNI load-unload for both centos and Cori */
+#ifndef CENTOS
     beforeLoadingGniDriverBlockAddressRanges(start1, end1, start2, end2);
+#endif
```

Comment:
I would suggest changing the design like this:
1. change the function name to `reserve_memory_regions`
2. In the implementation of this function, use `#ifdef` to have different implementations for different cases,
```
#ifdef GNI
// reserve the memroy regions
#endif
// do nothing for other cases
```
This would be much more cleaner and general, and easier to expand if we need to add a new case.
https://en.wikipedia.org/wiki/Open–closed_principle

Reply:
yes, i was speaking to twinkle about this, and this function should be working on CentOS as well. i'll try to debug it with the goal of generalizing it such that we don't require an `#ifdef` clause.